### ISSUE TITLE
Photos uploaded with "/api/v1/media" or "api/media/upload" remain private

### ISSUE THREAD

**Issue by SpcCw** [friendica/friendica#10993]
- [ Y ] I have searched open and closed issues for duplicates

### Bug Description

Trying to work around #10990 I switched to `api/media/upload` call to upload images. However it doesn't work for me either - pictures do get uploaded but after making the post they don't lose "Yourself" view permission [like they did in the past](https://github.com/friendica/friendica/issues/8477#issuecomment-643791747).

### Steps to Reproduce

1.  Upload an image using `api/media/upload`
2.  Make a public post with the image.
3.  Check the image permissions or log off and look at your public post.

Actual Result:

The image stays private.

Expected Result:

The image should be public. Also the setting "Make all posted pictures accessible" seems to be not working anymore so direct links don't work as well?

Example:
https://friends.deko.cloud/photo/38609214116195232f59d31348988916-0.jpg

### Platform Info

Friendica Version:

Current develop.

**Comment by SpcCw** [comment]
I noticed that this API call seems to be deprecated and the new one is `/api/v1/media` so I tried it instead.

I can post images with it but still they end up private (with only "Yourself" permission) even when used in public post. The option "Make all posted pictures accessible" also has no effect on these images.

**Comment by annando** [comment]
I'm unsure if this can be called a bug. Friendica will change the permissions for photos, once they are published in a post. However, we could extend this API call, so that a permission could be set as well.

**Comment by SpcCw** [comment]
@annando thanks for getting back to this old issue!

The problem was with permissions not being assigned at all, i.e. when I posted something with `/api/v1/media`  then the photo was successfully uploaded but had 'Yourself' permission and making the post did not change it so it remained inaccessible. It used to work before but it got broken at some point.

However I checked it now again and currently it works correctly - photos are assigned correct permissions after posting so it was fixed. No work on this is needed.
